providers/redhat/package_feed.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redhat
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/facebookincubator/nvdtools/providers/redhat/schema"
+	"github.com/facebookincubator/nvdtools/rpm"
+	"github.com/facebookincubator/nvdtools/wfn"
+	"github.com/pkg/errors"
+)
+
+// PackageFeed is an association between package names and the list of CVEs,
+// fixed or not, that have been recording against that package.
+// Packages are identified by their base package names without any epoch,
+// version, release.
+type PackageFeed map[string][]*schema.CVE
+
+// addPackage adds pkg to the list of packages only if not already there. A map
+// is probably too much overhead for this.
+func addPackage(pkgs []string, pkg string) []string {
+	for _, p := range pkgs {
+		if p == pkg {
+			return pkgs
+		}
+	}
+	return append(pkgs, pkg)
+}
+
+// PackageFeed transforms a Feed into a PackageFeed.
+func (feed Feed) PackageFeed() PackageFeed {
+	pkgFeed := PackageFeed{}
+
+	for _, cve := range feed {
+		pkgs := []string{}
```

Comment:
Looks weird, I see `pkgs := make([]string, 0)` syntax more often and `var pkgs []string` even more often. Why would you need to allocate an empty slice?